microprofile/server/src/main/java/io/helidon/microprofile/server/JaxRsService.java

```diff
@@ -201,6 +202,7 @@ private void handle(ServerRequest req, ServerResponse res) {
     }
 
     private void doHandle(Context ctx, ServerRequest req, ServerResponse res) {
+        ServerResponseHeaders savedResponseHeaders = ServerResponseHeaders.create(res.headers());
```

Comment:
Could this be moved to right before the `routing.reset()` ?  That would avoid the deep copy of headers until we know we need it.

Reply:
I concluded we have to save them early because `routing.reset()` clears the response headers (which is what raised this issue in the first place). 

It would need to do that only if some code in Jersey's attempt to process the request (filters? other code?) had _set_ some response headers that, now that we realize Jersey _did not_ actually work on that request, need to be _unset_. 

So what needs to happen there is to restore the response headers to their state before Jersey had a chance to change them.